### ISSUE TITLE
DirectX9 backend with minimimized viewport can assert

### ISSUE THREAD

**Issue by mq1n** [ocornut/imgui#3424]
**Version/Branch of Dear ImGui:**

Version: 1.79
Branch: docking
Full build info: https://gist.github.com/mq1n/5d74b0b889eac7cf9602e09729039028

**Back-end/Renderer/Compiler/OS**

Back-ends: imgui_impl_dx9cpp + imgui_impl_win32.cpp
Compiler: Visual studio 2019 Win32
Operating System: Windows 10 (1909)

**My Issue/Question:**

- Enable viewport on rendering ('io.ConfigViewportsNoAutoMerge' on demo app)
- Open ImGui app and keep in foreground
- Launch a new random process as admin(need uac confirmation screen spawn)
- Crash/assertion on ImGui app,.

**Screenshots/Video**

https://streamable.com/e4zdt3

**Comment by ocornut** [comment]
While investigating #6887 i stumbled on this same issue and applied the fix you suggest with cbc474d. Thank you!

**Comment by sakiodre** [comment]
While having a similar issue, I discovered that the proper method for managing HRESULT is to employ the SUCCEEDED or FAILED macro. From [Error Handling in COM](https://learn.microsoft.com/en-us/windows/win32/learnwin32/error-handling-in-com):

> The constants S_OK and S_FALSE are both success codes. Probably 99% of COM methods return S_OK when they succeed; but do not let this fact mislead you. A method might return other success codes, so always test for errors by using the SUCCEEDED or FAILED macro.

Upon examining our code base, I found 112 instances where `== S_OK` or `!= S_OK` is used, which is not the correct approach.

Initially, I wanted to propose using the appropriate macro. However, the descriptions for other success codes beginning with "S_" are somewhat unclear. I'm uncertain if there have been instances where a function completely failed but still returned an "S_" code.
